File: 2nd year/MPP/project-mpp/java-project/TriathlonServices/src/main/java/triathlon/services/ITriathlonServices.java
package triathlon.services;

import triathlon.model.Athlete;
import triathlon.model.Game;
import triathlon.model.Referee;
import triathlon.model.Result;

import java.util.Collection;
import java.util.Map;

public interface ITriathlonServices {
    Collection<Athlete> getAthletes() throws TriathlonException;

    Map<String, Float> getAthletesTotalPoints() throws TriathlonException;

    Game getGameById(Long id) throws TriathlonException;

    Referee authenticate(Referee referee, ITriathlonObserver client) throws TriathlonException;

    void logout(Referee referee, ITriathlonObserver client) throws TriathlonException;

    Collection<Result> getResultsForGame(Long gameId) throws TriathlonException;

    void setResult(Athlete athlete, Game game, Float value) throws TriathlonException;
}

File: 2nd year/MPP/project-mpp/csharp-project/services/ITriathlonServices.cs
using System.Collections.Generic;
using model;

namespace services
{
    public interface ITriathlonServices
    {
        IList<Athlete> GetAthletes();

        Dictionary<string, double> GetAthletesTotalPoints();

        Game GetGameById(long id);

        Referee Authenticate(Referee referee, ITriathlonObserver client);

        void Logout(Referee referee, ITriathlonObserver client);

        IList<Result> GetResultsForGame(long gameId);

        void SetResult(Athlete athlete, Game game, double value);
    }
}
